### PR TITLE
plan: resolve indices for apply schema

### PR DESCRIPTION
**Pull request by coocood**
If the apply schema is not resolved, the handle will be lost, then update will update the row with 0 handle.

### CODE REVIEW

**Comment by coocood** [comment]
@zz-jason @winoros PTAL

File: executor/write_test.go [hunk 1]
```diff
@@ -1594,3 +1594,16 @@ func (s *testSuite) TestDataTooLongErrMsg(c *C) {
...
+func (s *testSuite) TestUpdateSelect(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("create table msg (id varchar(8), b int, status int, primary key (id, b))")
+	tk.MustExec("insert msg values ('abc', 1, 1)")
+	tk.MustExec("create table detail (id varchar(8), start varchar(8), status int, index idx_start(start))")
+	tk.MustExec("insert detail values ('abc', '123', 2)")
+	tk.MustExec(`
+UPDATE msg SET msg.status = (SELECT detail.status FROM detail WHERE detail.start = '123' AND msg.id = detail.id)
+WHERE EXISTS (SELECT 1 FROM detail d WHERE  d.start = '123' AND msg.id = d.id)`)
```
Comment (zimulala) [inline executor/write_test.go]: Could we remove line 1607? I think it makes the test easier

**Pull request merged**